### PR TITLE
Added change_map service to map_server

### LINKED ISSUE #279: map_server: reload_map_from_file service?
Hi there,

I was wondering if it would be useful to implement a service on map_server to reload the map from a file. I think it would be beneficial in multi-floor environments where one needs to change the map due to the robot changing floors, instead of having to kill the  map_server node and starting it again with the map of the new floor.

Please let me know if this can be already achieved by some other means

### PR DESCRIPTION
**Pull request by sloretz**
This PR adds a "change_map" service to map_server including a change_map rostest. It also resolves #279.

The change_map service expects a path to a yaml file, and responds with a Boolean indicating if the map was successfully loaded. The new map and metadata are published to the appropriate topics. 

If change_map is called with the currently loaded map, the current map is reloaded as if it were a brand new map, resolving #279.

### CODE REVIEW

**Comment by sloretz** [comment]
CI failed with an unusual error. Is the build system misconfigured?

```
$ rake
rake aborted!
No Rakefile found (looking for: rakefile, Rakefile, rakefile.rb, Rakefile.rb)
/home/travis/.rvm/gems/ruby-1.9.3-p551/bin/ruby_executable_hooks:15:in `eval'
/home/travis/.rvm/gems/ruby-1.9.3-p551/bin/ruby_executable_hooks:15:in `<main>'
(See full trace by running task with --trace)


The command "rake" exited with 1.
```

File: map_server/srv/LoadMap.srv [hunk 1]
```diff
@@ -0,0 +1,3 @@
+string map_path
+---
+bool success
```
Comment (mikeferguson) [inline map_server/srv/LoadMap.srv:3]: I'd really rather we not put messages in the packages in navigation --  we have a separate repo (https://github.com/ros-planning/navigation_msgs) that is for messages, this has the advantage of making the interface itself (the messages) lightweight to build, especially if someone is on an alternate platform where they can't actually build/install the full navigation stack, but they just want to communicate with it

Comment (mikeferguson) [inline map_server/srv/LoadMap.srv:3]: Also, the right place for this message is probably nav_msgs, given that is where GetMap is, etc (although I do realize that is in common_msgs, meaning getting a review/release is likely going to take longer)

Comment (sloretz) [inline map_server/srv/LoadMap.srv:3]: I can submit a PR to common_msgs/nav_msgs for LoadMap.srv. Do you think the service would pass scrutiny in that repo?

Would that mean a path for this change to be released is to wait for the service to be released in common_msgs, then modify this PR to not contain the LoadMap service?

**Comment by DLu** [comment]
I'm digging up old PRs. This is the oldest. Can you rebase this on melodic?

**Comment by DLu** [comment]
Actually, this is blocked by https://github.com/ros/common_msgs/pull/95

**Comment by DLu** [comment]
Now blocked by https://github.com/ros/common_msgs/pull/152

**Comment by DLu** [comment]
No longer blocked.

**Comment by sloretz** [comment]
Closing in favor of #1029 :)

**Pull request closed without merging**